resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/ProxyInvocationHandler.java

```diff
@@ -162,23 +163,26 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
             try {
                 return method.invoke(target, args);
             } catch (InvocationTargetException e) {
-                if (e.getCause() instanceof ExceptionMapping.HandlerException) {
```

Comment:
Not a concern for this PR, but wondering if there's a way to consolidate `ProxyInvocationHandler` and `InvocationContextImpl` as the code seems quite similar at first glance

Reply:
Very similar, and I actually think those branches cannot get any `CompletionException` but added it just in case. They have slightly different cases though, and one even closes something, so I didn't dare touch them.

Reply:
not a problem, just curious